src/Angor/Client/Pages/Invest.razor

```diff
@@ -769,6 +770,37 @@
         }
     }
 
+    private async Task SaveInvestmentsListToNostrAsync()
+    {
+        var words = await passwordComponent.GetWalletAsync();
+        var rootNostrPrivateKey = await _derivationOperations.DeriveProjectNostrPrivateKeyAsync(words, 0); 
+        var rootNostrPrivateKeyHex = Encoders.Hex.EncodeData(rootNostrPrivateKey.ToBytes());
+
+        var nostrDMKey = _derivationOperations.DeriveNostrPubKey(words, 1);
+
+        Investments investments = new()
+        {
+            ProjectIdentifiers = storage.GetInvestmentProjects()
+                .Where(x => x.InvestedInProject())
+                .Select(x => x.ProjectInfo.ProjectIdentifier)
+                .ToList()
+        };
+            
+        var encryptedProjectIdList = await javascriptNostrToolsModule.InvokeAsync<string>(
+            "encryptNostr",
+            rootNostrPrivateKeyHex,
+            nostrDMKey,
+            System.Text.Json.JsonSerializer.Serialize(investments));
+            
+        _RelayService.SendDirectMessagesForPubKeyAsync(rootNostrPrivateKeyHex, nostrDMKey, encryptedProjectIdList, x =>
+        {
+            if (!x.Accepted)
+            {
+                //TODO need to handle failure to save the list of projects
```

Comment:
How do we do that error message to the user?

Reply:
Lets have a meeting about this..